workflows/detaxizer.nf

```diff
@@ -105,95 +119,175 @@ workflow DETAXIZER {
-    //
-    ch_parsed_kraken2_report = PARSE_KRAKEN2REPORT.out.to_filter.map {meta, path -> path}
+        //
+        // MODULE: Run Kraken2
+        //
+        KRAKEN2_KRAKEN2 (
+            ch_fastq_for_classification,
+            KRAKEN2PREPARATION.out.db.first(),
+            params.save_output_fastqs,
+            true
+        )
+        ch_versions = ch_versions.mix(KRAKEN2_KRAKEN2.out.versions.first())
 
-    KRAKEN2_KRAKEN2.out.classified_reads_assignment.combine(ch_parsed_kraken2_report).set{ ch_combined }
 
-    ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN (
-        ch_combined
-    )
+        //
+        // MODULE: Parse the taxonomy from the kraken2 report and return all subclasses of the tax2filter
+        //
+        PARSE_KRAKEN2REPORT(
+            KRAKEN2_KRAKEN2.out.report.take(1)
+        )
+        ch_versions = ch_versions.mix(PARSE_KRAKEN2REPORT.out.versions)
 
-    ch_versions = ch_versions.mix(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.versions.first())
 
-    //
-    // MODULE: Summarize the kraken2 results and the isolated kraken2 hits
-    //
-    ch_prepare_summary_kraken2 = KRAKEN2_KRAKEN2.out.classified_reads_assignment.join(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.classified).map {
-        meta, path1, path2 ->
-            return [ meta, [ path1, path2 ] ]
+        //
+        // MODULE: Isolate the hits for a certain taxa and subclasses
+        //
+        ch_parsed_kraken2_report = PARSE_KRAKEN2REPORT.out.to_filter.map {meta, path -> path}
+
+        KRAKEN2_KRAKEN2.out.classified_reads_assignment.combine(ch_parsed_kraken2_report).set{ ch_combined }
+
+        ISOLATE_KRAKEN2_IDS (
+            ch_combined
+        )
+
+        ch_versions = ch_versions.mix(ISOLATE_KRAKEN2_IDS.out.versions.first())
+
     }
 
-    ch_combined_kraken2 = ch_prepare_summary_kraken2.map {
-        meta, path ->
-            return [ meta +[ id: meta.id.replaceAll("(_R1|_R2)", "") ] , path]
-        }
-        .map {
-            meta, path ->
-                path = path.flatten()
-                return [meta, path]
+    if (params.classification_bbduk) {
+
+
+        //
+        // MODULE: Run bbduk
+        //
+        BBMAP_BBDUK (
+            ch_fastq_for_classification,
+            fasta_bbduk.first()
+        )
+        ch_versions = ch_versions.mix(BBMAP_BBDUK.out.versions.first())
+
+
+        //
+        // MODULE: Run ISOLATE_BBDUK_IDS
+        //
+        ISOLATE_BBDUK_IDS(
+            BBMAP_BBDUK.out.contaminated_reads
+        )
+        ch_versions = ch_versions.mix(ISOLATE_BBDUK_IDS.out.versions.first())
+
+
+    }
+
+    // Prepare MERGE_IDS Channel (with or without merging of IDs)
+
+    if (params.classification_bbduk && params.classification_kraken2){
+
+
+        //
+        // MODULE: Merge IDs
+        //
+        MERGE_IDS(
+            ISOLATE_KRAKEN2_IDS.out.classified_ids.join(
+                ISOLATE_BBDUK_IDS.out.classified_ids, by: [0]
+            ).map{
```

Comment:
What is the purpose of this map? You don't assign the output to a channel for downstream use (unless this is something about Nextflow I don't know?)

Reply:
The map produces a single list of input files compared to 2 separate files produced by join. This is necessary for the ID merger to work with either case: single file input or multi file input.

Reply:
Ah ok, understood! I always misremember between with `join` and `combine` :see_no_evil: